packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -166,12 +166,19 @@ export const vModelRadio: ModelDirective<HTMLInputElement> = {
 }
 
 export const vModelSelect: ModelDirective<HTMLSelectElement> = {
-  created(el, binding, vnode) {
+  created(el, { modifiers: { number } }, vnode) {
     addEventListener(el, 'change', () => {
+      const castToNumber = number
       const selectedVal = Array.prototype.filter
         .call(el.options, (o: HTMLOptionElement) => o.selected)
         .map(getValue)
-      el._assign(el.multiple ? selectedVal : selectedVal[0])
+
+      let domValue: string | number = el.multiple ? selectedVal : selectedVal[0]
+
+      if (castToNumber) {
+        domValue = toNumber(domValue)
+      }
```

Comment:
@edison1105, I would suggest handling the case of multiple select here.
When the `el.multiple` is true, `domValue` will be assigned with an array from `selectedVal`.
The `domValue = toNumber(domValue)` will cast an array of numbers to number.

This will make the multiple select case throw an error of:
```
<select multiple v-model> expects an Array or Set value for its binding, but got Number.
```

Reply:
Thanks~ @69hunter. I will update this.